stratum/hal/lib/nikss/nikss_wrapper.cc

```diff
@@ -0,0 +1,76 @@
+#include "stratum/hal/lib/nikss/nikss_wrapper.h"
+
+#include <memory>
+#include <set>
+#include <utility>
+#include <fstream>
+#include <string>
+#include <iostream>
+
+#include "absl/memory/memory.h"
+#include "absl/synchronization/mutex.h"
+#include "stratum/glue/status/status.h"
+#include "stratum/lib/utils.h"
+#include "stratum/lib/macros.h"
+
+extern "C" {
+#include "nikss/nikss.h"
+#include "nikss/nikss_pipeline.h"
+}
+
+// A macro for simplify checking the return value of NIKSS API.
+// For now, we always return ERR_INTERNAL.
+#define RETURN_IF_NIKSS_ERROR(expr)                                                \
+  do {                                                                       \
+    /* Using _status below to avoid capture problems if expr is "status". */ \
+    const int __ret = (expr);                                   \
+    if (__ret != 0) {                                 \
+      return MAKE_ERROR(ERR_INTERNAL) << "Return Error: "                        \
+                                      << #expr << " failed with code " << __ret; \
+    }                                                                            \
+  } while (0)
+
+
+namespace stratum {
+namespace hal {
+namespace nikss {
+
+NikssWrapper* NikssWrapper::singleton_ = nullptr;
+ABSL_CONST_INIT absl::Mutex NikssWrapper::init_lock_(absl::kConstInit);
+
+NikssWrapper::NikssWrapper() {}
+
+::util::Status NikssWrapper::AddPipeline(int pipeline_id,
+                                         const std::string bpf_obj) {
+  std::string tmp_filepath = "/etc/stratum/bpf.o";
```

Comment:
If the file is expected to be only temporary, it will be a lot of better to store it under `/tmp` directory (e.g. `/tmp/stratum/bpf.o`